### PR TITLE
Cuda11 syntax fixes

### PR DESCRIPTION
**Pull request by wgarunap**
- syntax fixes added

### CODE REVIEW

File: dnn/generated_algorithmperformance.go [hunk 1]
```diff
@@ -11,13 +11,13 @@ type AlgorithmPerformance struct {
...
 	algoDesc *AlgorithmDescriptor
-	status   Status
+	status   *cudnnStatus
 	time     float32
 	memory   uintptr
 }
 
 // NewAlgorithmPerformance creates a new AlgorithmPerformance.
-func NewAlgorithmPerformance(algoDesc *AlgorithmDescriptor, status Status, time float32, memory uintptr) (retVal *AlgorithmPerformance, err error) {
+func NewAlgorithmPerformance(algoDesc *AlgorithmDescriptor, status *cudnnStatus, time float32, memory uintptr) (retVal *AlgorithmPerformance, err error) {
```
Comment (chewxy) [inline dnn/generated_algorithmperformance.go:20]: This should not change. This should be `Status`. Remember you don't want people who are using the library to suddenly have to drop in to type `C.XXX`

Comment (chewxy) [reply dnn/generated_algorithmperformance.go:20]: Actually, we should create a type alias `type Status = cudnnStatus`

File: dnn/generated_algorithmperformance.go [hunk 2]
```diff
@@ -38,14 +38,14 @@ func NewAlgorithmPerformance(algoDesc *AlgorithmDescriptor, status Status, time
...
-func (a *AlgorithmPerformance) C() C.cudnnAlgorithmPerformance_t { return a.algoPerf }
+// C returns the cgo representation.
+func (a *AlgorithmPerformance) C() C.cudnnAlgorithmPerformance_t { return a.internal }
 
 // AlgoDesc returns the internal algoDesc.
 func (a *AlgorithmPerformance) AlgoDesc() *AlgorithmDescriptor { return a.algoDesc }
 
 // Status returns the internal status.
-func (a *AlgorithmPerformance) Status() Status { return a.status }
+func (a *AlgorithmPerformance) Status() *cudnnStatus { return a.status }
```
Comment (chewxy) [inline dnn/generated_algorithmperformance.go:48]: Should not be a pointer

**Review by chewxy — COMMENTED** [review]
very very excellent!

File: dnn/status.go [hunk 3]
```diff
@@ -5,8 +5,9 @@ import "C"
 
 type cudnnStatus int
 
-func (err cudnnStatus) Error() string  { return err.String() }
-func (err cudnnStatus) String() string { return resString[err] }
+func (err cudnnStatus) Error() string      { return err.String() }
+func (err cudnnStatus) String() string     { return resString[err] }
+func (err cudnnStatus) C() C.cudnnStatus_t { return err }
```
Comment (chewxy) [inline dnn/status.go:10]: this is wrong. This should be `return C.cudnnStatus_t(err)`

File: dnn/handle.go [hunk 4]
```diff
@@ -30,15 +30,15 @@ func NewContext() (retVal *Context) {
 }
 
 //  Close destroys the underlying context.
-func (ctx *Context) Close() error {
+func (co *Context) Close() error {
```
Comment (chewxy) [inline dnn/handle.go:33]: can we stick to using `ctx` for the variable name?

**Comment by wgarunap** [comment]
@chewxy 
Thanks you so much for the feedback, as per the review I have made changes on the code, please review new changes.

File: dnn/generated_algorithmdescriptor.go [hunk 5]
```diff
@@ -10,18 +10,37 @@ import "runtime"
 type AlgorithmDescriptor struct {
 	internal C.cudnnAlgorithmDescriptor_t
 
-	algorithm TODO
+	algorithm interface{}
```
Comment (chewxy) [inline dnn/generated_algorithmdescriptor.go:13]: Should we name this interface, and give dummy methods to things that implement it? Thoughts?

Comment (wgarunap) [reply dnn/generated_algorithmdescriptor.go]: It's good to hide the type in AlgorithmDescriptor, so in future we can use union replacement if any :)

File: dnn/generated_API.go [hunk 6]
```diff
@@ -144,7 +144,7 @@ func (co *Context) BatchNormalizationForwardTraining(mode BatchNormMode, alpha f
...
 }
-func (co *Context) CTCLoss(probsDesc *TensorDescriptor, probs Memory, hostLabels TODO, hostLabelLengths TODO, hostInputLengths TODO, costs Memory, gradientsDesc *TensorDescriptor, gradients Memory, algo CTCLossAlgo, ctcLossDesc *CTCLoss, workspace Memory, workSpaceSizeInBytes uintptr) error {
+func (co *Context) CTCLoss(probsDesc *TensorDescriptor, probs Memory, hostLabels []int, hostLabelLengths []int, hostInputLengths []int, costs Memory, gradientsDesc *TensorDescriptor, gradients Memory, algo CTCLossAlgo, ctcLossDesc *CTCLoss, workspace Memory, workSpaceSizeInBytes uintptr) error {
 	// call cudnnCTCLoss
 	return result(C.cudnnCTCLoss(co.internal, probsDesc.internal, unsafe.Pointer(probs.Uintptr()), hostLabels, hostLabelLengths, hostInputLengths, unsafe.Pointer(costs.Uintptr()), gradientsDesc.internal, unsafe.Pointer(gradients.Uintptr()), algo.C(), ctcLossDesc.internal, unsafe.Pointer(workspace.Uintptr()), C.size_t(workSpaceSizeInBytes)))
```
Comment (chewxy) [inline dnn/generated_API.go:149]: FYI, this will fail: `hostLabels` is `[]int`, which will need to be converted into `C.int*`. Use `ints2CInptPtr`

**Pull request merged**